app/assets/javascripts/discourse/app/controllers/full-page-search.js

```diff
@@ -308,6 +315,34 @@ export default Controller.extend({
       : "search-info";
   },
 
+  @discourseComputed("model.posts")
+  searchResultPosts(posts) {
+    if (additionalSearchResults.length > 0) {
+      // Apply RRF to results from each source:
+      const mixedResults = [
+        ...reciprocalRankingAlgorithm(posts),
+        ...reciprocalRankingAlgorithm(additionalSearchResults),
+      ];
+
+      // Sort results by RRF
+      const sortedResults = sortByReciprocalRanking(mixedResults);
+
+      // Remove duplicate results
```

Comment:
Since reciprocal ranking only sorts results, it may make sense to remove duplicates before mixing in and sorting the additional items.